src/box/sql/mem.c

```diff
@@ -1366,8 +1366,10 @@ mem_copy(struct Mem *to, const struct Mem *from)
 	if ((to->flags & MEM_Zero) != 0)
 		return sqlVdbeMemExpandBlob(to);
 	to->zMalloc = sqlDbReallocOrFree(to->db, to->zMalloc, to->n);
-	if (to->zMalloc == NULL)
+	if (to->zMalloc == NULL) {
```

Comment:
Why you do not check the case when `n == 0`? In this case it is not an error to get NULL from realloc(), I believe.

Reply:
For later usage (and avoid 1 assert elsewhere we have to set `szMalloc` to 0 here, we can't just simply bail out in this case without setting this field)

Reply:
That is not a proper reason, I think.